### ISSUE TITLE
Single line comments in JSON5 formatted presets are not supported for bitbucket-server platform

### ISSUE THREAD

**Issue by agriessbosch** [renovatebot/renovate#17321]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.66.2

### Please select which platform you are using if self-hosting.

Bitbucket Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

v7.6.16

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I have configuration presets stored which contain single-line comments, an example would be this (repo & path: renovate-presets/reviewers.json5):
```
{
   // add reviewers
   "reviewers": ["user1", "user2"]
}
```
This preset is referred to like this:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>PROJECT/renovate-presets/reviewers.json5",
  ]
}
```
The preset fails to parse, indicating broken JSON content.
If I wrap the comment in /* */ it works as expected.

Suspected root-cause is the merging of the retrieved lines from bitbucket-server which drops new line characters done here: https://github.com/renovatebot/renovate/blob/main/lib/config/presets/bitbucket-server/index.ts#L49

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Preset fetch error (repository=PROJECT/renovate-playground)
       "preset": "local>PROJECT/renovate-presets/reviewers.json5",
       "err": {
         "message": "invalid preset JSON",
         "stack": "Error: invalid preset JSON\n    at parsePreset (/usr/src/app/node_modules/renovate/lib/config/presets/util.ts:87:11)\n    at fetchJSONFile (/usr/src/app/node_modules/renovate/lib/config/presets/bitbucket-server/index.ts:49:21)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at fetchPreset (/usr/src/app/node_modules/renovate/lib/config/presets/util.ts:53:19)\n    at getPreset (/usr/src/app/node_modules/renovate/lib/config/presets/index.ts:221:22)\n    at Object.resolveConfigPresets (/usr/src/app/node_modules/renovate/lib/config/presets/index.ts:298:27)\n    at mergeRenovateConfig (/usr/src/app/node_modules/renovate/lib/workers/repository/init/merge.ts:215:5)\n    at getRepoConfig (/usr/src/app/node_modules/renovate/lib/workers/repository/init/config.ts:16:12)\n    at initRepo (/usr/src/app/node_modules/renovate/lib/workers/repository/init/index.ts:34:12)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:40:14)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:150:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       }
 INFO: Throwing preset error (repository=PROJECT/renovate-playground)
       "validationError": "Preset is invalid JSON (local>PROJECT/renovate-presets/reviewers.json5)"
 INFO: Repository has invalid config (repository=PROJECT/renovate-playground)
       "error": {
         "validationError": "Preset is invalid JSON (local>PROJECT/renovate-presets/reviewers.json5)",
         "message": "config-validation",
         "stack": "Error: config-validation\n    at Object.resolveConfigPresets (/usr/src/app/node_modules/renovate/lib/config/presets/index.ts:309:25)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at mergeRenovateConfig (/usr/src/app/node_modules/renovate/lib/workers/repository/init/merge.ts:215:5)\n    at getRepoConfig (/usr/src/app/node_modules/renovate/lib/workers/repository/init/config.ts:16:12)\n    at initRepo (/usr/src/app/node_modules/renovate/lib/workers/repository/init/index.ts:34:12)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:40:14)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:150:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       }
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
feel free to send a PR to join with `\n` 🤗

**Comment by HonkingGoose** [comment]
> ### If you're self-hosting Renovate, tell us what version of Renovate you run.
> 
> 32.66.2

The latest release for Renovate is `32.170.0`. You should update to get new features and fixes. 😉

**Comment by agriessbosch** [comment]
Will do both. Just waiting on internal processes to complete so I can sign the CLA and send the PR.

**Comment by viceice** [comment]
Please also add a test 😉

**Comment by agriessbosch** [comment]
Will do so. While I'm still waiting for approval, I discovered that the same issue exists when retrieving the renovate.json5 configuration file for a repository when `useBaseBranchConfig=merge` is set and baseBranches are used (see https://github.com/renovatebot/renovate/blob/9f837f226d5718a5fd7445d0367c1d3e30488c03/lib/modules/platform/bitbucket-server/index.ts#L135)

May I adapt the issue description or do I have to raise a new issue for that?

**Comment by viceice** [comment]
leave issue as is and just open a single PR to fix both issues
